File: articles/knowledge-architecture-and-systems-thinking/fortran/systems_knowledge_architecture_metrics.f90
program systems_knowledge_architecture_metrics
  implicit none

  real :: objects
  real :: relationships

  objects = 12.0
  relationships = 14.0

  print *, "Systems-Oriented Knowledge Architecture Metrics"
  print *, "Metadata coverage: ", 11.0 / objects
  print *, "Feedback role coverage: ", 8.0 / objects
  print *, "Relationship traceability: ", 13.0 / relationships
  print *, "Feedback edge share: ", 6.0 / relationships
  print *, "Underspecified relationship risk: ", 1.0 / relationships
end program systems_knowledge_architecture_metrics

File: articles/knowledge-architecture-and-systems-thinking/cpp/systems_knowledge_architecture_metrics.cpp
#include <iostream>
#include <iomanip>

int main() {
    double objects = 12.0;
    double relationships = 14.0;

    std::cout << std::fixed << std::setprecision(3);
    std::cout << "Metadata coverage: " << 11.0 / objects << "\n";
    std::cout << "Feedback role coverage: " << 8.0 / objects << "\n";
    std::cout << "Relationship traceability: " << 13.0 / relationships << "\n";
    std::cout << "Feedback edge share: " << 6.0 / relationships << "\n";
    std::cout << "Underspecified relationship risk: " << 1.0 / relationships << "\n";

    return 0;
}
